File: extension/src/capabilities/health.ts
import type { CapabilityHealthStatus, CapabilityManifestEntry } from './manifest'

export interface CapabilityHealth {
  id: string
  status: CapabilityHealthStatus
  reason?: string
}

export function summarizeCapabilityHealth(capabilities: CapabilityManifestEntry[]): CapabilityHealth[] {
  return capabilities.map((capability) => ({
    id: capability.id,
    status: capability.health,
  }))
}

File: backend/app/capability_platform/health.py
from __future__ import annotations

from app.contracts.capabilities import CapabilityHealth


def available_health(run_id: str = "capability-platform") -> CapabilityHealth:
    return CapabilityHealth(run_id=run_id, status="available")


def degraded_health(reason: str, run_id: str = "capability-platform") -> CapabilityHealth:
    return CapabilityHealth(run_id=run_id, status="degraded", reason=reason)
